dvc/repo/plots/__init__.py

```diff
@@ -486,7 +486,7 @@ def _collect_pipeline_files(repo, targets: List[str], props, onerror=None):
         for elem in plots_def:
             if isinstance(elem, str):
                 dvcfile_defs_dict[elem] = None
-            else:
+            elif elem:
```

Comment:
do we still need this then? I would rather put an assert here ...

Reply:
I prefer to be defensive in the parsing code as possible. If we had type information from schema (like #9606), we'd know what they really are. But voluptuous just gives us a blob of dictionary that I like to be defensive.

Reply:
Either works. Can go with assert too.